storage/plugins/oai-pmh-server/OaiPmhServerPlugin.php

```
@@ -0,0 +1,2410 @@
+            }
+        }
+
+        // Sort merged results by datestamp, then apply cursor + limit.
+        usort($rows, function (array $a, array $b): int {
+            $da = (string) ($a['_datestamp'] ?? '');
+            $db = (string) ($b['_datestamp'] ?? '');
+            return strcmp($da, $db) ?: (int) ($a['id'] ?? 0) - (int) ($b['id'] ?? 0);
+        });
+
+        return array_slice($rows, $cursor, $limit);
+    }
```

PHPMD (2.15.0)

[warning] The method fetchRecordsPage() has a Cyclomatic Complexity of 37. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)